js/script.js

```diff
@@ -290,26 +320,83 @@ const createTaskElement = (task) => {
 };
 
 const searchTaskHandler = () => {
-    $searchInput.focus();
+    let searchedTasks = searchTasks($searchInput.value);
+
+    if (filterBy !== ALL) {
+        searchedTasks = filterTasks(searchedTasks);
+    }
 
-    const searchedTask = sanitizeInput($searchInput.value).toLowerCase();
+    renderTasks(searchedTasks);
+};
 
-    const searchedTasks = tasks.filter((task) =>
+const searchTasks = (searchInput) => {
+    const searchedTask = sanitizeInput(searchInput).toLowerCase();
+
+    return tasks.filter((task) =>
         task.title.toLowerCase().includes(searchedTask)
     );
+};
 
-    renderTasks(searchedTasks);
+const handleSearchButton = () => {
+    $searchInput.value = "";
+
+    $searchInput.hidden = !$searchInput.hidden;
+
+    if (!$searchInput.hidden) {
+        $searchInput.focus();
+    } else {
+        handlePageTasks();
+    }
```

Comment:
Let's use early return here.

```suggestion
    if (!$searchInput.hidden) {
        $searchInput.focus();
        
        return;
    }
    
    handlePageTasks();
```